### PR TITLE
Add optional query accounting

### PR DESCRIPTION
**Pull request by cdegroot**
When you set `query_accounting` to true in your database.yml, then lines like

```
ACCT 6 bytes from SHOW TABLES LIKE 'users' 
```
will be emitted on stdout. These can be used to tally up frequency and bandwidtth usage by query.

### CODE REVIEW

**Comment by cdegroot** [comment]
(note that our branching model is screwed; 0.3.21.pd.3 is what we currently run but somehow we need to turn this into 0.3.21.pd.4; merging this PR onto its base branch is probably not what we want; so I guess that pushing this onto a new branch with the correct version name is the way forward)

File: ext/mysql2/client.c [hunk 1]
```diff
@@ -127,6 +127,31 @@ static void rb_mysql_client_mark(void * wrapper) {
...
+      return 0;
+  }
+  if (result->data == NULL) {
+      return 0;
+  }
+  if (result->data->data == NULL) {
+      return 0;
+  }
+  unsigned long size = 0;
+  for (MYSQL_ROWS *row = result->data->data; row != NULL; row = row->next) {
```
Comment (burnison) [inline ext/mysql2/client.c]: I could be overlooking something, but I _think_ you can just sum up [`length` in `st_mysql_rows`](https://github.com/percona/percona-server/blob/mysql-5.5.41/include/mysql.h#L141). Or maybe even better, [`lengths` in the `result`](https://github.com/percona/percona-server/blob/mysql-5.5.41/include/mysql.h#L292). That said, I have no clue if it works as expected.

**Comment by cdegroot** [comment]
I thought about such edge cases, and decided to let it run and see what happens from some real load. Nothing should crash, at the very least :). Thanks for the review!

**Comment by cdegroot** [comment]
(these commits are going onto the 0.3.21.pd.4 branch, closing without merging therefore. Not ideal...)

**Pull request closed without merging**